cli/src/semgrep/commands/scan.py

```diff
@@ -712,6 +711,8 @@ def scan(
             "Cannot create auto config when metrics are off. Please allow metrics or run with a specific config."
         )
 
+    jobs = jobs if jobs else 1 if deep else __get_cpu_count()
```

Comment:
I think this will only set the job to 1 when running `semgrep scan --deep` but not `semgrep ci` and will also avoid setting jobs to 1 https://github.com/returntocorp/semgrep/pull/6624/files

Might be worth setting this further down and closer to deepsemgrep invocation?